storage/storage-client/src/lib.rs

```diff
@@ -214,7 +214,7 @@ impl StorageRead for StorageReadServiceClient {
         &self,
         start_epoch: u64,
         end_epoch: u64,
-    ) -> Pin<Box<dyn Future<Output = Result<Vec<LedgerInfoWithSignatures>>> + Send>> {
+    ) -> Pin<Box<dyn Future<Output = Result<(Vec<LedgerInfoWithSignatures>, bool)>> + Send>> {
```

Comment:
I don't realize this is not returning a `ValidatorChangeEventWithProof` we probably should embed the bool into that struct too, so that the receiver of that can understand it needs to iterate through

Reply:
I think that is the plan, but it seems to require more changes, so probably a separate PR.

Reply:
make sense